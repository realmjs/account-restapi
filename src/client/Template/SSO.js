"use strict"

import React, { PureComponent } from 'react'

export default class extends PureComponent {
  constructor(props) {
    super(props)
  }
  componentDidMount() {
    this.props.done({ status: 200, session: window__data.session })
  }
  render() {
    return( <div />)
  }
}
