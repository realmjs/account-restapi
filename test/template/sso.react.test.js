"use strict"

import React from 'react';
import renderer from 'react-test-renderer';

import SSO from '../../src/client/template/SSO';

beforeEach(() => window.__data = { status: 200, session: 'user-session' });

test('Should match snapshot and call done when mounted', () => {
  const done = jest.fn(msg => msg);
  const component = renderer.create(
    <SSO done = {done} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(done).toHaveBeenCalled();
  expect(done.mock.results[0].value).toEqual({ status: 200, session: 'user-session'});
});
