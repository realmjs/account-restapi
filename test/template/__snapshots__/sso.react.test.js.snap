// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot and call done when mounted 1`] = `<div />`;
